### PR TITLE
[#1139] added better locking to document based dirty resource

### PR DESCRIPTION
**Pull request by cdietrich**
[#1139] added better locking to document based dirty resource
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java [hunk 1]
```diff
@@ -479,13 +479,16 @@ public void markEditorClean(IDirtyStateEditorSupportClient client) {
...
-		if (!getDirtyResource().isInitialized())
-			return;
-		for(IResourceDescription.Delta delta: event.getDeltas()) {
-			if (delta.getOld() == getDirtyResource().getDescription() || delta.getNew() == getDirtyResource().getDescription())
+		DocumentBasedDirtyResource resource = getDirtyResource();
+		synchronized(resource) {
+			if (!resource.isInitialized())
 				return;
+			for(IResourceDescription.Delta delta: event.getDeltas()) {
+				if (delta.getOld() == resource.getDescription() || delta.getNew() == resource.getDescription())
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java]: i am not sure if it is better to store resource.getDescription in a variable and end synchronization here

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java]: Yes, that would make sense indeed. The description should be a copy and it *should* be ok to work with that one in the loop without a sync block.

**Comment by szarnekow** [comment]
Approved given the change you proposed within `descriptionsChanged`

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/DirtyStateEditorSupport.java]: @szarnekow updated pr. ok now?

**Comment by szarnekow** [comment]
LGTM

**Pull request merged**